lib/SimplifyPointerBitcastPass.cpp

```diff
@@ -759,7 +759,7 @@ bool clspv::SimplifyPointerBitcastPass::runOnPHIFromGEP(Module &M) const {
   bool changed = false;
   DenseMap<Value *, Type *> type_cache;
 
-  SmallVector<std::pair<GetElementPtrInst *, Type *>> Worklist;
+  DenseMap<GetElementPtrInst *, Type *> Worklist;
```

Comment:
This makes iteration non-deterministic. Since we iterate through the worklist below this might change code generation.

Suggest using either UniqueVector or an explicit parallel map.

Reply:
Let's say there is a block with 2 phis for pointers we are processing. Is it not possible modifications will occur that change the relative order of instructions based on which phi is processed first?